src/Core/Client/Endpoint.php

```diff
@@ -194,9 +194,13 @@ public function getCore(): ?string
      * @param int $timeout
      *
      * @return self Provides fluent interface
+     *
+     * @deprecated Endpoint::setTimeout is deprecated since Solarium 5.2 and will be removed in Solarium 6. Configure the timeout on the HTTP Client used by the Adapter instead.
      */
     public function setTimeout(int $timeout): self
     {
+        @trigger_error('Endpoint::setTimeout is deprecated since Solarium 5.2 and will be removed in Solarium 6. Configure the timeout on the HTTP Client used by the Adapter instead.', E_USER_DEPRECATED);
```

Comment:
Shall we keep this deprecation then?

The timeout is currently used in the non-deprecated Adapters `Curl` and `Http`... so for those there is no upgrade path yet

Reply:
I suggest to keep the deprecation here. It will be kind of inconvienient for users of the curl of adapter. Therefore we should mention in the documentation how to port your code to solarium 6 (not now).

Reply:
But on Solarium 5.2 I cannot get rid of this deprecation which is not good. We should already have an alternative then.

Reply:
You're right.

In the future people would have to adjust timeouts directly in the component they inject, for example guzzle.

Let's follow this path. I suggest to copy the timeout getter an setter to the curl and http themselves. In the Client interface we already have getAdapter(). So if you use curl or http adapters you can get rid of the deprecation by calling `$client->getAdapter()->setTimeout()`.

Reply:
@mkalkbrenner I propose to do the endpoint timeout deprecation in a follow-up PR. I will work on it once this one is approved :wink: